FWCore/Framework/test/stubs/ToyIntProducers.cc

```diff
@@ -486,6 +486,100 @@ namespace edmtest {
+                                                             edm::EndRunProducer> {
+  public:
+    explicit NonEventIntProducer(edm::ParameterSet const& p)
+        : brToken_{produces<IntProduct, edm::Transition::BeginRun>("beginRun")},
+          blToken_{produces<IntProduct, edm::Transition::BeginLuminosityBlock>("beginLumi")},
+          elToken_{produces<IntProduct, edm::Transition::EndLuminosityBlock>("endLumi")},
+          erToken_{produces<IntProduct, edm::Transition::EndRun>("endRun")},
+          value_(p.getParameter<int>("ivalue")) {
+      {
+        auto tag = p.getParameter<edm::InputTag>("consumesBeginRun");
+        if (not tag.label().empty()) {
+          brGet_ = consumes<IntProduct, edm::InRun>(tag);
+        }
+      }
+      {
+        auto tag = p.getParameter<edm::InputTag>("consumesBeginLuminosityBlock");
+        if (not tag.label().empty()) {
+          blGet_ = consumes<IntProduct, edm::InLumi>(tag);
+        }
+      }
+      {
+        auto tag = p.getParameter<edm::InputTag>("consumesEndLuminosityBlock");
+        if (not tag.label().empty()) {
+          elGet_ = consumes<IntProduct, edm::InLumi>(tag);
+        }
+      }
+      {
+        auto tag = p.getParameter<edm::InputTag>("consumesEndRun");
+        if (not tag.label().empty()) {
+          erGet_ = consumes<IntProduct, edm::InRun>(tag);
+        }
+      }
+    }
+    void accumulate(edm::StreamID iID, edm::Event const& e, edm::EventSetup const& c) const override;
+    void globalBeginRunProduce(edm::Run& e, edm::EventSetup const&) const override;
+    void globalEndRunProduce(edm::Run& e, edm::EventSetup const&) const override;
+    void globalBeginLuminosityBlockProduce(edm::LuminosityBlock& e, edm::EventSetup const&) const override;
+    void globalEndLuminosityBlockProduce(edm::LuminosityBlock& e, edm::EventSetup const&) const override;
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& conf) {
+      edm::ParameterSetDescription desc;
+      desc.add<int>("ivalue");
+      desc.add<edm::InputTag>("consumesBeginRun", {});
+      desc.add<edm::InputTag>("consumesEndRun", {});
+      desc.add<edm::InputTag>("consumesBeginLuminosityBlock", {});
+      desc.add<edm::InputTag>("consumesEndLuminosityBlock", {});
+
+      conf.addDefault(desc);
+    }
+
+  private:
+    const edm::EDPutTokenT<IntProduct> brToken_;
+    const edm::EDPutTokenT<IntProduct> blToken_;
+    const edm::EDPutTokenT<IntProduct> elToken_;
+    const edm::EDPutTokenT<IntProduct> erToken_;
+    const int value_;
+    edm::EDGetTokenT<IntProduct> brGet_;
+    edm::EDGetTokenT<IntProduct> blGet_;
+    edm::EDGetTokenT<IntProduct> elGet_;
+    edm::EDGetTokenT<IntProduct> erGet_;
+  };
+
+  void NonEventIntProducer::accumulate(edm::StreamID iID, edm::Event const& e, edm::EventSetup const&) const {}
+  void NonEventIntProducer::globalBeginRunProduce(edm::Run& r, edm::EventSetup const&) const {
+    if (not brGet_.isUninitialized()) {
+      r.get(brGet_);
```

Comment:
Would it be worth of the effort to check that the read value actually is what is expected?

Reply:
There are many other tests which check that one gets back the actual data one requested. We could do the same here. I'm OK either way.